viamrobotics/rdk
# RSDK-1808: Create classifications transform

This PR adds a classifications transform to be used in the transform pipeline for transform cameras. 

**Manual Testing**
I manually tested this on a fake robot powered by my laptop. See the screen recording below:

https://user-images.githubusercontent.com/9754276/217956641-d1e5b2cd-7ec5-4d48-9f62-95621420821a.mov

For the automated tests, I set up a transform camera with a classifications stage in its pipeline, reading from a source camera that reads a static file, which in this case is an image of a keyboard. It then reads the image from the transform camera and tests if the color red is present at a specific coordinate in the upper-left-hand corner of the image. I also wrote the entire image to a file to ensure it was what I expected, and I've included that below. The text is a little bit smaller on this image because the source image is larger; I'm not sure if this is a problem for overlaying the text on the image coming out of a real camera, but I mostly followed what was done for the bounding boxes.

![with_label](https://user-images.githubusercontent.com/9754276/217957216-69222109-16a6-41b8-8bb1-b53de431bfa4.jpg)

This is the updated image after I received some feedback to: 1.) make the confidence score a percentage with fewer significant digits and 2.) allow for multiple classifications overlaid on the image.
![with_label](https://user-images.githubusercontent.com/9754276/218793227-9164d199-3862-46bb-9259-070b29b5c9f3.jpg)


**Open Questions**
1. Right now I'm displaying at most one label/confidence score pair on the image (none if it cannot confidently classify anything in the image). Do we want more than those or is one enough? 
**SOLUTION:** I've made max_classifications configurable, so now users can request more than one classification.
2. Question above ^ about the size of the text relative to the image.
**SOLUTION**: Filed a ticket to address this for both classifications and detections [here](https://viam.atlassian.net/browse/RSDK-1917).
